test/evp_kdf_test.c

```diff
@@ -251,6 +314,86 @@ static int test_kdf_hkdf_invalid_digest(void)
     return ret;
 }
 
+static int test_kdf_hkdf_derive_set_params_fail(void)
+{
+    int ret = 0, i = 0;
+    EVP_KDF_CTX *kctx = NULL;
+    OSSL_PARAM params[2];
+    unsigned char out[10];
+
+    if (!TEST_ptr(kctx = get_kdfbyname(OSSL_KDF_NAME_HKDF)))
+        goto end;
+    /* Set the wrong type for the digest so that it causes a failure */
+    params[0] = OSSL_PARAM_construct_int(OSSL_KDF_PARAM_DIGEST, &i);
+    params[1] = OSSL_PARAM_construct_end();
+    if (!TEST_int_eq(EVP_KDF_derive(kctx, out, sizeof(out), params), 0))
```

Comment:
Would a set_params be a better choice here?

The derive call can fail for other reasons than the one being tested.  Without having set anything up, it will fail even if the param is accepted.

Reply:
No I am trying to hit a failure point in the derive function.

Reply:
The comment is misleading then...

Reply:
??. The derive() internally calls set params which fails and then returns 0.